### PR TITLE
feat(vscode): make quick-lint a jerk 🖕

### LINKED ISSUE #1188: VSCode: support snarky option
related: #529 #530

### PR DESCRIPTION
**Pull request by vegerot**
feat(vscode): make quick-lint a jerk 🖕

Summary: It ticks me off 😠 how polite 😇 quick-lint is when I code.  If I mess
up 😬, I want it to tell me.  Stop beating around the bush 🌳 and tell it to me
like it is. 💯

Test plan: 📝
1. open TypeScript file with no errors in it ✅
2. add an error ❌
3. note how quick-lint tells you what the problem is like a sissy 😭
4. enable settings > quick-lint-js > snarky 😈
5. note how quick-lint is straight up with you.  You suck. 🗑️

Closes #1188

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/quick-lint/quick-lint-js/pull/1190).
* __->__ #1190
* #1189

### CODE REVIEW

File: plugin/vscode/test/vscode-tests.js [hunk 1]
```diff
@@ -1095,6 +1095,196 @@ tests = {
...
+        {
+          code: {
+            target: "https://quick-lint-js.com/errors/E0164/",
+            value: "E0164",
+          },
+          message: "yeah, JSON sucks; try quick-lint-json",
+        },
+      ]
+    );
+  },
```
Comment (vegerot) [inline plugin/vscode/test/vscode-tests.js:1287]: I should probably add `enable_snarky_config_file` and `disable_snarky_config_file` tests

Comment (strager) [reply plugin/vscode/test/vscode-tests.js:1287]: I think you can add quick-lint-js.config stuff into the other three tests.

Alternatively, you could data-drive the tests (see line 24 for an existing example):
```javascript
for (let testCase of [
  {
    fileName: 'hello.js',
    content: 'undeclaredVariable',
    englishMessage: 'use of undeclared variable: undeclaredVariable',
    snarkyEnglishMessage: 'did you fail spelling class?',
  },
  {
    fileName: 'quick-lint-js.config',
    content: '{',
    englishMessage: '...',
    snarkyEnglishMessage: 'yeah, JSON sucks; try quick-lint-json',
  },
]) {
  tests = {
    ...tests,
    [`enabling snarky re-lints (${testCase.fileName})`]: async ({ addCleanup }) => {
      ...
    },
  };
}
```

**Review by strager — CHANGES_REQUESTED** [review]
Must fix: Remove profanity.

**Comment by vegerot** [comment]
> Must fix: Remove profanity.

is 🖕🏾 "profanity"?

File: plugin/vscode/test/vscode-tests.js (hunk 1 above)
Comment (vegerot) [reply plugin/vscode/test/vscode-tests.js]: That's an awesome idea 💡.  This is why they pay you the big bucks

**Comment by strager** [comment]
Landed as Git commit 86bb25fd928fe78f162fdaa4ff4f3f3e9c7f4b8c.

File: plugin/vscode/quick-lint-js/vscode/qljs-workspace.cpp [hunk 2]
```diff
@@ -234,10 +243,29 @@ ::Napi::Value QLJS_Workspace::configuration_changed(
...
+
+  this->is_snarky_enabled_ = is_snarky;
+  if (is_snarky) {
+    this->translator_.use_messages_from_locale("en_US@snarky");
+  } else {
+    // TODO(#529): Use the locale from the VS Code configuration.
+    this->translator_.use_messages_from_source_code();
+  }
+  this->qljs_documents_.for_each(
+      [this, is_snarky, env](::Napi::Value value) -> void {
```
Comment (strager) [inline plugin/vscode/quick-lint-js/vscode/qljs-workspace.cpp:263]: Compiler was warning about `is_snarky` being an unused capture. I fixed it by using `[&]` instead.

**Pull request closed without merging**